tests/integration/issues/github_976/test_topk_rest_api.py

```diff
@@ -24,6 +24,6 @@ def test_top_k_with_rest_api(query_dict):
         req = request.Request(f'http://0.0.0.0:{port}/api/search', data=query,
                               headers={'content-type': 'application/json'})
         resp = request.urlopen(req).read().decode('utf8')
-        assert json.loads(resp)['queryset'][0]['name'] == 'VectorSearchDriver'
-        assert json.loads(resp)['queryset'][0]['parameters']['top_k'] == TOP_K
+        assert json.loads(resp)['queryset'][0]['name'] == 'SliceQL'
```

Comment:
the user then cant use a SliceQL in the middle of the Flow. that can make for a very hard to debug problem

Reply:
the current `VectorSearchDriver` does not garantee `top_k`, it is wrongly implemented. jina hello-world says `topk=50` but the result only shows 20.

Reply:
I'm very surprised how these two past issues #1330 #1316 are considered as done.

Reply:
yes, but because under our current implementation is very hard to guarantee this. We cannot target a specific Pod to send top_k. in my opinion, current solution is the best tradeoff.

But anyway, It is important to know that no solution right now will be perfect

Reply:
if the user does not use `SliceQL` then there is no way to select top_k.

Reply:
when we first implemented top_k we were aware that it was just syntax sugar to help the first user, but that if someone wanted to have full control of the parameters they should send their own QueryLang with the Queries.

Thats why the issues were considered done, because the issues came from a degradation of that original thought.